ddl/ddl_api.go

```diff
@@ -658,6 +658,42 @@ func checkColumnsAttributes(colDefs []*ast.ColumnDef) error {
 	return nil
 }
 
+// checkColumnFieldLength check the maximum length limit for different character set varchar type columns.
+func checkColumnFieldLength(schema *model.DBInfo, colDefs []*ast.ColumnDef, tbInfo *model.TableInfo) error {
+	for _, colDef := range colDefs {
+		if colDef.Tp.Tp == mysql.TypeVarchar {
+			var setCharset string
+			setCharset = mysql.DefaultCharset
+			if len(schema.Charset) != 0 {
+				setCharset = schema.Charset
+			}
+			if len(tbInfo.Charset) != 0 {
+				setCharset = tbInfo.Charset
+			}
+
```

Comment:
Please add some comment to explain that `colDef.Tp.Charset` will be check in `planner/core/preprocess.go`, So ignore here.

Reply:
I got some problem here:
In mysql: 
```sql
mysql root@127.0.0.1:test> create table t1 (a varchar(20000) ) default charset utf8;
Query OK, 0 rows affected
Time: 0.008s
mysql root@127.0.0.1:test> show create table t1
+-------+--------------------------------------+
| Table | Create Table                         |
+-------+--------------------------------------+
| t1    | CREATE TABLE `t1` (                  |
|       |   `a` varchar(20000) DEFAULT NULL    |
|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+-------+--------------------------------------+
1 row in set
Time: 0.003s
mysql root@127.0.0.1:test> alter table t1 convert to character set utf8mb4;
(1074, u"Column length too big for column 'a' (max = 16383); use BLOB or TEXT instead")
```

In TiDB This PR:
```sql
mysql root@127.0.0.1:test> create table t1 (a varchar(20000) ) default charset utf8;
Query OK, 0 rows affected
Time: 0.005s
mysql root@127.0.0.1:test> alter table t1 convert to character set utf8mb4;
Query OK, 0 rows affected
Time: 0.005s
mysql root@127.0.0.1:test> show create table t1
+-------+-------------------------------------------------------------+
| Table | Create Table                                                |
+-------+-------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (                                         |
|       |   `a` varchar(20000) DEFAULT NULL                           |
|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------+
```